### ISSUE TITLE
[Awaiting payment 22nd Jan] [$2000] Text duplicated when using voice dictation in form fields (i.e profile > display name)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17174]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Click on settings
2. Click on profile
3. Click on edit name
4. Click the native Apple microphone icon and dictate a first name
5. Click the native Apple microphone again to end the dictation session

## Expected Result:
The word dictated should only appear once as text. 

## Actual Result:
The text gets duplicated

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native - iPhone SE2020
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.98-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/230698594-7ce80ab1-2697-46f4-8318-455d5205a993.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @esh-g 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680908051128159

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f3efde20809bda43</li>
        <li>Upwork Job ID: 1646235992926674944</li>
        <li>Last Price Increase: 2023-05-03</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @trjExpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by luacmartins** [comment]
Doesn't seem to be specific to Form, since I see the same behavior with `OptionsSelector` in the `NewChatPage` for example

**Comment by trjExpensify** [comment]
>since I see the same behavior with OptionsSelector in the NewChatPage for example

Interesting. I can't reproduce the duplication on iOS native in the OptionSelector, only in the forms. Got a vid? 

https://user-images.githubusercontent.com/16232057/231466321-bc248061-a768-444e-a25c-9da1b39f7f93.MP4

**Comment by trjExpensify** [comment]
Ah hm, what's the diff? Are you clicking the microphone button in the bottom right or that tooltip popover thing?

Anyhow, it seems like this needs to be investigated and it can be external.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01f3efde20809bda43

**Comment by MelvinBot** [comment]
Current assignee @trjExpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @fedirjh (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Beamanator (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by luacmartins** [comment]
I tap the bottom right microphone in the keyboard and then it shows up again inline to stop the recording, which is what duplicates the message.

**Comment by trjExpensify** [comment]
>It seems an upstream bug

Ohh.. @hoangzinh do you have a link to that upstream?

**Comment by trjExpensify** [comment]
>@trjExpensify can I share it here or slack?

You can share it here for sure.

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by Beamanator** [comment]
@trjExpensify seems we might want to increase the bounty here?

**Comment by MelvinBot** [comment]
@Beamanator @trjExpensify @fedirjh this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by Beamanator** [comment]
Quick bump @trjExpensify can we double the bounty here?

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by trjExpensify** [comment]
Hey guys, I've unexpectedly been out of office. Upped the bounty to $2,000.

**Comment by ShogunFire** [comment]
Looks like it is a bug in IOS that may have something to do with accessibility: 
https://www.applevis.com/forum/ios-ipados/double-dictation-bug

https://apple.stackexchange.com/questions/370316/ios-dictation-duplicated

I tried to see if there is an event that would be triggered when the user use the dictation that could allow us to remove the duplicate but I didn't find it. For now I don't think we can do much

**Comment by Beamanator** [comment]
Thanks for the investigation @ShogunFire ! 

@trjExpensify what if we keep this bug open for a bit longer to see if anyone gets crazy ideas how to fix it (doubling along the way) and if we don't get takers by week 4 or 5 we close this out? (Since I don't know if it's worth trying to find a fix internally)

**Comment by fabriziobertoglio1987** [comment]
I was not able to reproduce it with react-native main on iOS 16.4 https://github.com/facebook/react-native/issues/36045#issuecomment-1529502317.

**Comment by trjExpensify** [comment]
>I was not able to reproduce it with react-native main on iOS 16.4 https://github.com/facebook/react-native/issues/36045#issuecomment-1529502317.

Ah, so is that to say that it's a bug in /App yeah?

**Comment by jpdriver** [comment]
hey folks 👋🏻 although the duplication doesn't happen every time a user tries to dictate (and @fabriziobertoglio1987 wasn't able to replicate it), I'm fairly confident it's an upstream bug in `react-native`..

fwiw I can reliable recreate it on iOS 16 (**outside of /App**); and I'm trying to investigate / debug further.

**Comment by MelvinBot** [comment]
📣 @jpdriver! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by fabriziobertoglio1987** [comment]
Thanks. I was able to reproduce on Expensify and react-native. I start investigating this issue. 

https://user-images.githubusercontent.com/24992535/236147203-876f83a8-0aa4-49f1-b6bc-c55678ea407c.mp4

https://user-images.githubusercontent.com/24992535/236148780-5511a6a7-5565-43d2-a89c-44d54d6f7f82.mp4

**Comment by Santhosh-Sellavel** [comment]
@Beamanator this one is being fixed along with other issue. #18121 we have an active PR against upstream

Can you put this on hold mean while, thanks!

**Comment by Beamanator** [comment]
@Santhosh-Sellavel are you sure this one will also get fixed by https://github.com/facebook/react-native/pull/37188? 😅 

I'm definitely happy to put this one on hold in the meantime 👍

**Comment by Santhosh-Sellavel** [comment]
Sorry for the delay @Beamanator yeah sure yes it seems to be handled there, please put this on hold meanwhile. 

cc: @hellohublot

**Comment by Beamanator** [comment]
@trjExpensify Just noting we're putting this on hold for https://github.com/Expensify/App/issues/18121, which is technically also on hold for https://github.com/facebook/react-native/pull/37188

**Comment by trjExpensify** [comment]
Cool, thanks! Upstream we go :)

**Comment by trjExpensify** [comment]
Same as 2 days ago Melv.

**Comment by Beamanator** [comment]
held PR (https://github.com/facebook/react-native/pull/37188) still on hold

**Comment by trjExpensify** [comment]
Merged, and now we're implementing in our fork https://github.com/Expensify/App/issues/18121#issuecomment-1567629728

**Comment by trjExpensify** [comment]
@fedirjh @Beamanator looks like we're still waiting on https://github.com/Expensify/App/issues/18121#issuecomment-1579480966

**Comment by trjExpensify** [comment]
Samesies, we need a RN release or merge in our fork.

**Comment by trjExpensify** [comment]
Awaiting the next RN release which includes the fix for this.

**Comment by trjExpensify** [comment]
No change!

**Comment by trjExpensify** [comment]
No change, waiting on the RN release still.

**Comment by trjExpensify** [comment]
Same, awaiting the next RN release which includes the fix for this.

**Comment by Beamanator** [comment]
Ok I thinkkk RN just got updated!

**Comment by trjExpensify** [comment]
To 0.73 which this is included in?

**Comment by Beamanator** [comment]
Hmm only to 0.72.3 actually, so seems we're still holding

**Comment by trjExpensify** [comment]
Cool, thanks!

**Comment by trjExpensify** [comment]
Still waiting, going to pop this on `monthly`.

**Comment by trjExpensify** [comment]
Samesies!

**Comment by trjExpensify** [comment]
@fedirjh where can we conclusively check the progress on the new release?

**Comment by trjExpensify** [comment]
Nice! Added that to the issue title for safe keeps.

**Comment by trjExpensify** [comment]
0.73 upgrade is in progress: [#31558](https://github.com/Expensify/App/pull/31558)

**Comment by trjExpensify** [comment]
Okay, #31558 is through the door. So I think we're done here? We only need to pay @esh-g $250 (grandfathered rate for the time of issue creation) for reporting the issue. Correct?

**Comment by trjExpensify** [comment]
@esh-g sent you an offer!

**Comment by esh-g** [comment]
@trjExpensify Sorry for the delay! Accepted the offer now

**Comment by trjExpensify** [comment]
Paid!
